### PR TITLE
fix: consolidate StripFrontmatter into shared FrontmatterUtility (#216)

### LINKED ISSUE #216: [P3] Consolidate StripFrontmatter implementations (tech debt)
## Context

**Ref:** AD-018, PR #200

Three separate `StripFrontmatter` implementations exist across different assemblies:
1. `ToolReader.StripFrontmatter()` — regex-based
2. `ComposedToolGeneratorService.StripFrontmatter()` — line-by-line
3. `PageGenerator.StripFrontmatter()` — line-by-line

AD-018 marked this as acceptable tech debt at project boundaries, but it should be consolidated when module boundaries are next refactored.

## Acceptance Criteria

- [ ] Single `StripFrontmatter` implementation in `DocGeneration.Core` or `DocGeneration.Core.NaturalLanguage`
- [ ] All three call sites updated to use the shared implementation
- [ ] Existing tests continue to pass
- [ ] No behavior change — pure refactor

## Owner

**Morgan** (C# Generator Developer) — owns the generator codebase.

### PR DESCRIPTION
**Pull request by diberry**
## Summary

Consolidates 4 independent `StripFrontmatter` implementations into a single shared method in `FrontmatterUtility` (`DocGeneration.Core.Shared`).

Closes #216

### New shared method
- **`FrontmatterUtility.StripFrontmatter()`** — compiled regex, handles null/empty, empty frontmatter, Windows line endings

### Updated call sites (4)
- `ComposedToolGeneratorService.cs` — delegates to shared method
- `PageGenerator.cs` — delegates to shared method (fully qualified to avoid conflict with local `FrontmatterUtility` wrapper)
- `ToolReader.cs` — delegates to shared method, removed `FrontmatterRegex` field
- `QualityMetrics.cs` — delegates to shared method, added `Core.Shared` ProjectReference

### Tests
- **Added** `FrontmatterUtilityStripTests.cs` with 10 tests (6 ported from `PageGeneratorStripFrontmatterTests` + 4 new edge cases)
- **Deleted** `PageGeneratorStripFrontmatterTests.cs` (superseded by shared tests)
- All 1,895 tests pass across 19 assemblies

## Testing
- `dotnet build docs-generation.sln` — 0 errors, 0 warnings
- `dotnet test docs-generation.sln` — all pass

### CODE REVIEW

File: docs-generation/DocGeneration.Core.Shared/FrontmatterUtility.cs [hunk 1]
```diff
@@ -165,4 +166,20 @@ public static string GenerateRawToolFrontmatter(string? mcpCliVersion, string? g
             generatedDate: generatedDate,
             msDate: generatedDate);
     }
+
+    private static readonly Regex StripFrontmatterRegex = new(
+        @"^---\s*\n(.*?\n)?---\s*\n?",
```
[the remark this answers is not in the document]
Comment (diberry) [reply docs-generation/DocGeneration.Core.Shared/FrontmatterUtility.cs]: Accepted the suggestion. Tightened regex tail from `\s*\n?` to `\r?\n?` to only consume the expected line ending after the closing `---`. Added two regression tests: one for body starting with indentation and one for body starting with a blank line.
